apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -254,6 +254,24 @@ public async Task<int> GetOrdinalAsync(string name, CancellationToken cancellati
             return _fieldIndex[name];
         }
 
+        /// <summary>
+        /// Gets the read timstamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was specified.
+        /// </summary>
+        /// <remarks>
+        /// The read timestamp is only available after at least one result is
+        /// read. If the returned results are empty, then a read timestamp
+        /// won't be available.
+        /// </remarks>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly becuase <see cref="TimestampBound.ReturnReadTimestamp" /> was not set.</exception>
+        public Timestamp GetReadTimestamp()
+        {
+            GaxPreconditions.CheckState(PopulateMetadata().Transaction != null, "Transaction unavailable so read timestamp can't be returned.");
+            GaxPreconditions.CheckState(PopulateMetadata().Transaction.ReadTimestamp != null, "ReadTimestamp is not returned in the transaction.");
```

Comment:
I'm not sure about this. If the timestamp can be unavailable because there was no data, I wonder whether returning a null timestamp would be better if this transaction just doesn't have a timestamp. Otherwise I can see it being tricky for users to know whether or not they can call this - I don't want every user to have to put this in a try/catch statement.

Reply:
Since it may be a bug in the client that we don't return a timestamp if there's no data (I still need to investigate and fix it), do you think we should still return null for the case where Read() or ReadAsync() has not yet been called?